lib/flash.js

```diff
@@ -0,0 +1,184 @@
+var async = require('async');
+var PAGE = 256;
+var logs = require('./logs');
+var CHIP_ID = new Buffer([0x01, 0x02, 0x19]);
+
+function address(addr) {
+  return [(addr >> 24) & 0xFF, (addr >> 16) & 0xFF, (addr >> 8) & 0xFF, addr & 0xFF];
+}
+
+// Generate a mediatek factory partition
+function factoryPartition(mac1, mac2) {
+  return concatToBuffer(
+    [0x20, 0x76, 0x03, 0x01],
+    mac1,
+    Array(30).fill(0xff),
+    mac2
+  );
+}
+
+function transaction(usbConnection, write, read, status_poll, wren, next) {
+  read = read || 0;
```

Comment:
The  argument value of this parameter is often `null`—it should just be a number all the time